src/foam/nanos/menu/Menu.js

```diff
@@ -70,13 +82,78 @@ foam.CLASS({
       name: 'icon',
       documentation: 'Icon associated to the menu item.',
       displayWidth: 80
+    },
+    {
+      documentation: 'Predicate providing arbitrary checks, in addition to the regular menu auth checks.',
+      class: 'FObjectProperty',
+      of: 'foam.mlang.predicate.Predicate',
```

Comment:
We can use PredicateProperty

    class: 'foam.mlang.predicate.PredicateProperty',